packages/discord.js/typings/index.test-d.ts

```diff
@@ -2370,33 +2371,33 @@ expectType<
 >(NonThreadGuildBasedChannel);
 expectType<GuildTextBasedChannel>(GuildTextBasedChannel);
 
-const button = new ButtonBuilder({
+const button = new PrimaryButtonBuilder({
   label: 'test',
   style: ButtonStyle.Primary,
-  customId: 'test',
+  custom_id: 'test',
 });
 
 const selectMenu = new StringSelectMenuBuilder({
-  maxValues: 10,
-  minValues: 2,
-  customId: 'test',
+  max_values: 10,
+  min_values: 2,
+  custom_id: 'test',
 });
 
 new ActionRowBuilder({
   components: [selectMenu.toJSON(), button.toJSON()],
 });
 
 new StringSelectMenuBuilder({
-  customId: 'foo',
+  custom_id: 'foo',
 });
 
-new ButtonBuilder({
+new PrimaryButtonBuilder({
   style: ButtonStyle.Danger,
 })
-  .setEmoji('<a:foo:123>')
-  .setEmoji('<:foo:123>')
-  .setEmoji('foobar:123')
-  .setEmoji('😏')
+  .setEmoji(resolvePartialEmoji('<a:foo:123>'))
+  .setEmoji(resolvePartialEmoji('<:foo:123>'))
+  .setEmoji(resolvePartialEmoji('foobar:123'))
+  .setEmoji(resolvePartialEmoji('😏'))
```

Comment:
If this will be the recommended way to do that in mainlib then the `resolvePartialEmoji` function should either be renamed to something easier and/or at least not be marked as private/internal in https://github.com/discordjs/discord.js/blob/39eadce9cbf28de9e8217056bc6aef41032f2ab9/packages/discord.js/src/util/Util.js#L115 and https://github.com/discordjs/discord.js/blob/39eadce9cbf28de9e8217056bc6aef41032f2ab9/packages/discord.js/typings/index.d.ts#L3406-L3408 anymore.

Reply:
Yeah for sure, all though for unicode emotes you can also do `{ name: '<emote>' }` very easily (on the topic of recommendations)